### ISSUE TITLE
[vdom] Rendering unescaped HTML produces non-functioning HTML input controls

### ISSUE THREAD

**Issue by eltimn** [marko-js/marko#629]
## Bug Report

I'm trying to use `$!{}` to output some HTML that is generated on the server, but it's not rendering.

### Context

I am passing an `input` tag like this: `<input type="text" value="" />`. Looking at the element in chrome devtools it looks like:  `<INPUT type="text" value=""></INPUT>`. If I edit the element in devtools to be like the input it works fine.

### Expected Behavior

It should output: `<input type="text" value="" />`

### Actual Behavior

It outputs:  `<INPUT type="text" value=""></INPUT>`
### Possible Fix

<details><summary>Additional Info</summary>

### Your Environment

* Version used: 4.1.3
* Environment name and version (e.g. Chrome 39, node.js 5.4): Node 6.9.2
* Operating System and version (desktop or mobile): Ubuntu 16.04

### Steps to Reproduce

Add the following component to your app:

```
class {

  onCreate() {
    this.state = {
      html: "<input type='text' name='test' value='x' />"
    }
  }
}

<div>$!{state.html}</div>
```

</details>

**Comment by eltimn** [comment]
A regular `div` tag works fine, although the tag is upper-cased. An `img` tag has the same problems as `input`, so it seems to be a problem with tags that don't require a closing tag.

**Comment by patrick-steele-idem** [comment]
Thanks for reporting the problem @eltimn. We found some issues when converting the VDOM nodes to actual DOM nodes for certain input controls. We put in some fixes that will go out with the upcoming v4.2.0 released scheduled for a little later today. Thanks, again.

**Comment by eltimn** [comment]
Excellent. Thank you.

**Comment by patrick-steele-idem** [comment]
New version published with fix: `marko@4.2.0`

Thanks, again.
